### DISCUSSION TITLE
Why does this.$persist not work within Alpine.data when using an arrow function?

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3526] [4. General]
`Alpine.data('$', () => ({ view: this.$persist('raster') }))` throws an error `_vscode_livepreview_injected_script:139 Alpine Expression Error: Cannot read properties of undefined (reading '$persist')`, while `Alpine.data('$', function(){ return { view: this.$persist('raster') }})` works.

Just wondering. Is this because `this` does works differently within arrow functions? Could you use `$persist` any other way inside the arrow function notation? Just wondering, because in the documentation, I always see the arrow function notation, never the regular function notation...

**Comment by ekwoka** [comment]
It has to do with the `this` of the context.

Really, just use `Alpine.$persist`.

The first one `this` is just referring to the global `this` so there is none of the Alpine magic there.

**Comment by simonferndriger** [comment]
`Alpine.$persist` works great
